### ISSUE TITLE
npm run generate works perfect but some nft's are skipped/not rendering

### ISSUE THREAD

**Issue by RequiemDreams** [HashLips/hashlips_art_engine#642]
I made sure all of my layers are formated properly and exported using the transparent frame for some reason when I npm run generate some of the NFT's in the build images file are skipped or not fully rendered see below

any suggestions are very much appreciated I've been at this for multiple days now.

![Screenshot (25)](https://user-images.githubusercontent.com/98503217/151784883-8a92227f-eb81-4dfa-a363-4399f8651459.png)
![237](https://user-images.githubusercontent.com/98503217/151784886-396d8ce6-aa2a-438c-bc3f-d36555173102.png)
![40](https://user-images.githubusercontent.com/98503217/151784890-97fe8f54-5e8b-4a48-9f97-79c2ea3f4f0e.png)
![791](https://user-images.githubusercontent.com/98503217/151784903-b9c03da6-ce36-4968-9e4d-0d7603583e30.png)

**Comment by bolshoytoster** [comment]
@RequiemDreams have you checked your `layers/` directory for any blank images, that could be it. If none of your background images are blank could you share your `src/config.js` file.

**Comment by RequiemDreams** [comment]
> @RequiemDreams have you checked your `layers/` directory for any blank images, that could be it. If none of your background images are blank could you share your `src/config.js` file.

@bolshoytoster it won't let me send config on here may I get your telegram Reddit or email address? thank you for taking the time to help :)

**Comment by bolshoytoster** [comment]
@RequiemDreams you can just copy/paste the contents.

**Comment by RequiemDreams** [comment]
@bolshoytoster 
below you can see I have 10,200 under grow addition size that was to account for unrendered images I was planning on deleting corrupted ones and changing all the numbers and json files to list 10k in sequence order this would take an eternity however so again I can't thank you enough for trying to help me.

const basePath = process.cwd();
const { MODE } = require(`${basePath}/constants/blend_mode.js`);
const { NETWORK } = require(`${basePath}/constants/network.js`);

const network = NETWORK.eth;

// General metadata for Ethereum
const namePrefix = "LostintheCosmos";
const description = "Inspired by the nations and corporations competing to reach limits in our universe never thought possible";
const baseUri = "ipfs://NewUriToReplace";

const solanaMetadata = {
  symbol: "YC",
  seller_fee_basis_points: 5, // Define how much % you want from secondary market sales 1000 = 10%
  external_url: "https://www.youtube.com/c/hashlipsnft",
  creators: [
    {
      address: "7fXNuer5sbZtaTEPhtJ5g5gNtuyRoKkvxdjEjEnPN4mC",
      share: 100,
    },
  ],
};

// If you have selected Solana then the collection starts from 0 automatically
const layerConfigurations = [
  {
    growEditionSizeTo: 10200,
    layersOrder: [
      { name: "Background" },
      { name: "behind the mask 5" },
      { name: "Flag Pole" },
      { name: "Glass" },
      { name: "OuterHelmet" },
      { name: "Hands" },
      { name: "Torso" },
      { name: "Feet" },
      { name: "Flag" },
      { name: "NameTags" },
    ],
  },
];

const shuffleLayerConfigurations = false;

const debugLogs = false;

const format = {
  width: 1080,
  height: 1080,
  smoothing: false,
};

const gif = {
  export: false,
  repeat: 0,
  quality: 100,
  delay: 500,
};

const text = {
  only: false,
  color: "#ffffff",
  size: 20,
  xGap: 40,
  yGap: 40,
  align: "left",
  baseline: "top",
  weight: "regular",
  family: "Courier",
  spacer: " => ",
};

const pixelFormat = {
  ratio: 2 / 128,
};

const background = {
  generate: true,
  brightness: "80%",
  static: false,
  default: "#000000",
};

const extraMetadata = {external_url: "https://www.lostinthecosmos.ca"
};

const rarityDelimiter = "#";

const uniqueDnaTorrance = 10000;

const preview = {
  thumbPerRow: 5,
  thumbWidth: 50,
  imageRatio: format.height / format.width,
  imageName: "preview.png",
};

const preview_gif = {
  numberOfImages: 5,
  order: "ASC", // ASC, DESC, MIXED
  repeat: 0,
  quality: 100,
  delay: 500,
  imageName: "preview.gif",
};

module.exports = {
  format,
  baseUri,
  description,
  background,
  uniqueDnaTorrance,
  layerConfigurations,
  rarityDelimiter,
  preview,
  shuffleLayerConfigurations,
  debugLogs,
  extraMetadata,
  pixelFormat,
  text,
  namePrefix,
  network,
  solanaMetadata,
  gif,
  preview_gif,
};

**Comment by bolshoytoster** [comment]
@RequiemDreams sorry for the late reply, I’m in school. If you look in the folder with the background images, is there a background that doesn’t appear in the results. If so that image is probably corrupted.

You could also try running it with only one image in the backgrounds folder and see if it still happens.

**Comment by RequiemDreams** [comment]
@bolshoytoster No apologies needed schools important,

I have confirmed that every image I have in my layers are showing up when I generate in a few different images. 

The problem seems to be occurring not from the exports themselves but during the generating process otherwise there’d be an image that didn’t show up in any of them. 

I tried running npm run generate after leaving just one background layer as you suggested and the issue still occurres in some images worse then others. 

There doesn’t seem to be any correlation or rhyme and reason to it either they just randomly don’t generate some layers selectively.

as image shows below some lose the head others lose the mask 

![image](https://user-images.githubusercontent.com/98503217/151797952-40f64812-e422-49c7-ae0f-edd168f06ced.jpg)

**Comment by RequiemDreams** [comment]
12/200 are like this

**Comment by bolshoytoster** [comment]
@RequiemDreams could you make a new GitHub repository and upload your project to it and I'll debug it after school (~2.5 hours.

**Comment by DiskJockey56** [comment]
I'm having the same issue only much worse. Most of mine aren't rendering fully and its completely random. when i check the json the traits are in there and sometimes they render most times they dont
Did you figure out the issue?

**Comment by DiskJockey56** [comment]
I'm also receiving this message after I run generate
Not sure if this is normal

(node:20506) [DEP0147] DeprecationWarning: In future versions of Node.js, fs.rmdir(path, { recursive: true }) will be removed. Use fs.rm(path, { recursive: true }) instead
(Use `node --trace-deprecation ...` to show where the warning was created)

**Comment by bolshoytoster** [comment]
@DiskJockey56 you can ignore that warning. Could you send your project (.zip) so I can see if I can recreate this.

**Comment by cf-devil-cry** [comment]
> I made sure all of my layers are formated properly and exported using the transparent frame for some reason when I npm run generate some of the NFT's in the build images file are skipped or not fully rendered see below
> 
> any suggestions are very much appreciated I've been at this for multiple days now.
> 
> ![Screenshot (25)](https://user-images.githubusercontent.com/98503217/151784883-8a92227f-eb81-4dfa-a363-4399f8651459.png)
> 
> ![101](https://user-images.githubusercontent.com/98503217/151785262-0c3b3bc5-87f4-4df4-b20a-0a139d739a10.png) ![102](https://user-images.githubusercontent.com/98503217/151785269-3423aad2-bf9f-4276-a89d-fb1cfe6fe750.png) ![162](https://user-images.githubusercontent.com/98503217/151785285-4eeb6933-6e55-496e-8d4c-017dc79d4ede.png)

Did you finally solve this problem? I also had the same problem？

**Comment by cf-devil-cry** [comment]
> @DiskJockey56 you can ignore that warning. Could you send your project (.zip) so I can see if I can recreate this.

Is there some news  in the problem?

**Comment by bolshoytoster** [comment]
@cf-devil-cry no, but could you send your project (either as a github repositiry or as a `.zip`) so I can see what causes this.

**Comment by DiskJockey56** [comment]
The problem for me was that I thought I was exporting transparent frames
but I was actually exporting white png frames through photoshop.
I had to use illustrator to create the blank frames cause photoshop wasn't
working everytime I went to export.
